@here/harp-mapview/lib/geometry/SolidLineMesh.ts

```diff
@@ -0,0 +1,451 @@
+        const endCapT = intersectClosestEndCap(ray, line, hWidth);
+
+        const minT = Math.min(extrLineT, endCapT);
+        if (minT === Infinity) {
+            return false;
+        }
+        ray.at(minT, interLine);
+        return true;
+    }
+
+    // The plain intersection is also a line intersection only if it's closer to the line
+    // than the extrusion half width.
+    const distSq = interPlane.distanceToSquared(line.closestPointToPoint(interPlane, true, tmpV4));
+
+    if (distSq > hWidthSq) {
+        return false;
+    }
+    interLine.copy(interPlane);
+    return true;
+}
+
+/**
+ * Finds the intersections of a ray with a partition of a solid line mesh representing a feature.
+ * @param mesh The mesh whose intersections will be found.
+ * @param raycaster Contains the intersection ray.
+ * @param localRay Same ray as raycaster.ray but in object local space.
+ * @param intersects Array that will contain all intersections found between ray and feature.
+ * @param halfWidth The line's extrusion half width.
+ * @param lHalfWidth The line's extrusion half width in mesh local space.
+ * @param lHalfWidthSq The line's extrusion half width squared in mesh local space.
+ * @param beginIdx The index where the feature starts in the mesh geometry's indices attribute.
+ * @param endIdx The index where the feature end in the mesh geometry's indices attribute.
+ * @param bSphere The feature bounding sphere.
+ * @returns feature
+ */
+function intersectFeature(
+    mesh: THREE.Mesh,
+    raycaster: THREE.Raycaster,
+    localRay: THREE.Ray,
+    intersects: THREE.Intersection[],
+    halfWidth: number,
+    lHalfWidth: number,
+    lHalfWidthSq: number,
+    beginIdx: number,
+    endIdx: number,
+    bSphere: THREE.Sphere
+): void {
+    const vExt = tmpV1;
+    const plane = tmpPlane;
+    const interPlane = tmpV2;
+    const line = tmpLine1;
+
+    const geometry = mesh.geometry as THREE.BufferGeometry;
+    const attributes = geometry.attributes;
+    const position = attributes.position as THREE.BufferAttribute;
+    const bitangent = attributes.bitangent;
+    const indices = geometry.index!.array;
+
+    tmpSphere.copy(bSphere);
+    tmpSphere.applyMatrix4(mesh.matrixWorld);
+    tmpSphere.radius += halfWidth;
+
+    if (!raycaster.ray.intersectsSphere(tmpSphere)) {
+        return;
+    }
+
+    for (let i = beginIdx; i < endIdx; i += SEGMENT_STRIDE) {
+        const a = indices[i];
+        const b = indices[i + VERTEX_STRIDE];
+
+        // Find the plane containing the line segment, using the segment start, end and extrusion
+        // vector.
+        line.start.fromBufferAttribute(position, a);
+        line.end.fromBufferAttribute(position, b);
+        vExt.set(bitangent.getX(a), bitangent.getY(a), bitangent.getZ(a)).normalize();
+        plane.setFromCoplanarPoints(line.start, tmpV3.copy(line.start).add(vExt), line.end);
+
+        // The ray intersection if any, will be on the extrusion plane.
```

Comment:
Based on that ... maybe we should not even use current intersectLine implementation which is more or less based on cylinder + caps (Capsule). If you want to search only point on the plane co-planar with ground / line surface then I would really simply entire test to check for plane -> ray intersection and then check intersection point distance from the line - this should be no bigger then halfWidth.

Reply:
That's exactly what I'm doing in case the ray is not coplanar to the extrusion plane. The cylinder+caps test is only used when the ray is coplanar to the extrusion plane.